### DISCUSSION TITLE
[Question] Is there a way to get the telemetry from the formation lap?

### DISCUSSION THREAD

**Discussion by samhoppen** [theOehrly/Fast-F1#416] [Q&A]
As the title states, is there a way to get the telemetry for drivers for the formation lap?

**Comment by theOehrly** [comment]
The telemetry data does exist, yes. But there's no clear indication in the data that tells you when the formation lap starts.
What you can do is to access the full telemetry data stream through ``Session.car_data`` (https://docs.fastf1.dev/core.html#fastf1.core.Session.car_data). The telemetry objects have a ``.slice_by_time`` method (https://docs.fastf1.dev/core.html#fastf1.core.Telemetry.slice_by_time). You can use ``Session.session_start_time`` as end time and then just subtract maybe 2 minutes from that for the start time of the formation lap.
If you want to get more accurate, you'd probably need to analyze the telemetry of the leaders car and check when the car starts moving.
